lib/logstash/outputs/elasticsearch/http_client/pool.rb

```diff
@@ -232,11 +232,18 @@ def health_check_request(url)
       perform_request_to_url(url, :head, @healthcheck_path)
     end
 
-    def healthcheck!
+    def healthcheck!(register_phase = true)
       # Try to keep locking granularity low such that we don't affect IO...
       @state_mutex.synchronize { @url_info.select {|url,meta| meta[:state] != :alive } }.each do |url,meta|
         begin
           health_check_request(url)
+
+          # when called from resurrectionist skip the product check done during register phase
+          if register_phase
```

Comment:
This makes me wonder if we could just make the `elasticsearch?` check everytime instead of HEAD'ing the `/`?

Right after we do HEAD then we request '/' in `es_version = get_es_version(url)`.

With this change we'll be doing 4 requests to ES:
1. HEAD request
2. GET '/' for `elasticsearch?` on `register_phase`
3. GET '/' to get elasticsearch version
4. GET '/_license'

We can just do instead:

1. GET '/' for healthcheck (and always confirm it's ES)
2. GET '/_license'

Reply:
I agree with you, I think that point 2 and 3 could be merged or at least do one one HTTP GET call to `/`.
About the HTTP HEAD, that call could use a customized path from configuration (`healthcheck_path`) that some users could have customized. If we drop the HEAD call is like ignoring that setting and we should remove it.
I'm ok with removing the HEAD call, because actually seems to be used just to ping the server for reachability and then it immediately send a GET.
We could split this work in 2 steps:
- first collapse the GET call for point 2 and 3
- create a follow-up PR to drop the HEAD call and deprecate the `healthcheck_path` param

Reply:
> create a follow-up PR to drop the HEAD call and deprecate the healthcheck_path param

It's fine if there's only an issue instead to track the goal of reducing the number of network calls for a healthcheck.